### PR TITLE
add cpack support to easily build .deb & .rpm packages

### PR DESCRIPTION
**Pull request by spoonincode**
Adds support for building .deb and .rpm packages (and binary tarballs). This is very basic support but I have validated that packages build and install cleanly.

I've never been able to figure out how to to get `GNUInstallDirs` and `cpack` to fully cooperate with each other (a bit of a conflict due to absolute vs relative paths in `install()` commands). So when building a package you probably want to `-DCMAKE_INSTALL_PREFIX=/usr` as I believe it's generally a faux pas to have a package place files in `/usr/local`.

~Also, I have only enabled tarballs by default since if I enable `TGZ;DEB;RPM` as the default users may get confused when they can't build an .rpm on Ubuntu (it does work if you have the tools installed, it just may be a little weirdly unexpected). Maybe the default should be based on the platform discovered during cmake configure or something?~ Now there is dpkg/rpmbuild detection.

So most likely, if you're building packages for Ubuntu, you'd want to do something like
```
cmake -DCMAKE_INSTALL_PREFIX=/usr ....
make package
```
I have _not_ made any changes to the CI to pop out packages yet pending further decisions.

.deb info for a build on Ubuntu 20.04
```
Architecture: amd64
Depends: libc6 (>= 2.29), libcurl4 (>= 7.16.2), libgcc-s1 (>= 3.4), libssl1.1 (>= 1.1.0), libstdc++6 (>= 9), libtinfo6 (>= 6), libusb-1.0-0 (>= 2:1.0.16), zlib1g (>= 1:1.2.0)
Description: daemon and CLI tools for Mandel blockchain protocol
Maintainer: EOS Network Foundation
Package: mandel
Priority: optional
Section: utils
Version: 3.1.0-dev
Installed-Size: 361276
```
Bundled files for said package
```
./usr/
./usr/bin/
./usr/bin/cleos
./usr/bin/eosio-blocklog
./usr/bin/keosd
./usr/bin/nodeos
./usr/bin/trace_api_util
./usr/share/
./usr/share/licenses/
./usr/share/licenses/eosio/
./usr/share/licenses/eosio/LICENSE
./usr/share/licenses/eosio/LICENSE.eos-vm
./usr/share/licenses/eosio/LICENSE.go
./usr/share/licenses/eosio/LICENSE.rapidjson
./usr/share/licenses/eosio/LICENSE.secp256k1
./usr/share/licenses/eosio/LICENSE.softfloat
./usr/share/licenses/eosio/LICENSE.wavm
./usr/share/licenses/eosio/LICENSE.yubihsm
```

.rpm info
```
Name        : mandel
Version     : 3.1.0_dev
Release     : 1
Architecture: x86_64
Install Date: (not installed)
Group       : unknown
Size        : 373390093
License     : unknown
Signature   : (none)
Source RPM  : mandel-3.1.0_dev-1.src.rpm
Build Date  : Thu 19 May 2022 01:30:22 PM EDT
Build Host  : zen
Vendor      : EOS Network Foundation
URL         : https://github.com/eosnetworkfoundation/mandel
Summary     : daemon and CLI tools for Mandel blockchain protocol
Description :
DESCRIPTION
===========

This is an installer created using CPack (https://cmake.org). No additional installation instructions provided.
```

### CODE REVIEW

**Review by jgiszczak — APPROVED** [review]
For future reference, the Debian Section should be `utils` rather than `devel`, for the package which corresponds to this tarball.  The package in the Debian repository which installs bitcoind is in `utils`.  A header package, if and when, should of course remain in `devel`.

File: package.cmake [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+set(CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}-${VERSION_FULL}")
+
+set(CPACK_PACKAGE_CONTACT "EOS Network Foundation")
+set(CPACK_PACKAGE_VENDOR "EOS Network Foundation")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "daemon and CLI tools for Mandel blockchain protocol")
+set(CPACK_PACKAGE_HOMEPAGE_URL "https://github.com/eosnetworkfoundation/mandel")
+
+set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+set(CPACK_DEBIAN_PACKAGE_SECTION "utils")
+
```
Comment (larryk85) [inline package.cmake:21]: Does this install the license files?

Comment (larryk85) [reply package.cmake:21]: For RPMs.

Comment (spoonincode) [reply package.cmake:21]:
```
$ rpm -qlp mandel-3.1.0-dev.rpm
/usr/bin/cleos
/usr/bin/eosio-blocklog
/usr/bin/keosd
/usr/bin/nodeos
/usr/bin/trace_api_util
/usr/lib/.build-id
/usr/lib/.build-id/36
/usr/lib/.build-id/36/2a0acc0ca38a6203fb2a1bc0d66cc6d633a151
/usr/lib/.build-id/4f
/usr/lib/.build-id/4f/94e7995aaefab8e75ad645c6a05b1692d15e43
/usr/lib/.build-id/96
/usr/lib/.build-id/96/ee6d87e466c6255f1ecc60d00188e3b6beea48
/usr/lib/.build-id/c2
/usr/lib/.build-id/c2/f8c4fcb52e805ae0553dcc7c6a8e212ef9e66d
/usr/lib/.build-id/c4
/usr/lib/.build-id/c4/57306c5df4b778e4d72caf99623bff5d2fb5e4
/usr/share/licenses
/usr/share/licenses/eosio
/usr/share/licenses/eosio/LICENSE
/usr/share/licenses/eosio/LICENSE.eos-vm
/usr/share/licenses/eosio/LICENSE.go
/usr/share/licenses/eosio/LICENSE.rapidjson
/usr/share/licenses/eosio/LICENSE.secp256k1
/usr/share/licenses/eosio/LICENSE.softfloat
/usr/share/licenses/eosio/LICENSE.wavm
/usr/share/licenses/eosio/LICENSE.yubihsm
```

**Review by larryk85 — APPROVED** [review]

**Pull request merged**